target/linux/ramips/files-4.14/drivers/net/ethernet/mediatek/mtk_eth_soc.c

```diff
@@ -565,7 +565,7 @@ static inline u32 fe_empty_txd(struct fe_tx_ring *ring)
 	barrier();
 	return (u32)(ring->tx_ring_size -
 			((ring->tx_next_idx - ring->tx_free_idx) &
-			 (ring->tx_ring_size - 1)));
```

Comment:
This is to make sure that we always leave one free tx descriptor in tx ring. This matches the behavior in upstream mtk_soc_eth:
https://elixir.bootlin.com/linux/latest/source/drivers/net/ethernet/mediatek/mtk_eth_soc.c#L1550
and also matches the behavior in raeth driver from ancient ralink linux SDK.
This has nothing to do with tx timeout issue and we'd better leave this one unchanged.